tests/sleepq_timed.c

```diff
@@ -0,0 +1,85 @@
+#include <sleepq.h>
+#include <ktest.h>
+#include <time.h>
+#include <thread.h>
+#include <sched.h>
+#include <runq.h>
+
+/* This test doesn't check if timed sleep is also indeed interruptible. */
+
+/* just to get some unique address for sleepq */
+static int wchan;
+
+#define T 6
+#define SLEEP_TIME_MS 4
+
+static volatile int timed_received;
+static volatile int signaled_received;
+
+static volatile int signaled_sent;
+
+static thread_t *waiters[T];
+static thread_t *waker;
+
+static void snorlax(void *_arg) {
+  systime_t before_sleep = getsystime();
+  sq_wakeup_t status = sleepq_wait_timed(&wchan, __caller(0), SLEEP_TIME_MS);
+  systime_t after_sleep = getsystime();
+  systime_t diff = after_sleep - before_sleep;
+
+  if (status == SQ_TIMEOUT) {
+    timed_received++;
+    assert(diff >= SLEEP_TIME_MS);
+  } else if (status == SQ_NORMAL) {
+    signaled_received++;
+  } else {
+    assert(false);
```

Comment:
Do not use such things in the code. There's `panic` procedure for that, which takes a message string.

Reply:
Done.